src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -597,6 +606,91 @@ private ProcessWaitState GetWaitState()
             return _waitStateHolder._state;
         }
 
+        private static (uint userId, uint groupId) GetUserAndGroupIds(ProcessStartInfo startInfo)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(startInfo.UserName));
+
+            (uint? userId, uint? groupId) = GetUserAndGroupIds(startInfo.UserName);
+
+            if (userId == null || groupId == null)
+            {
+                throw new Win32Exception(SR.Format(SR.UserDoesNotExist, startInfo.UserName));
+            }
+
+            return (userId.Value, groupId.Value);
+        }
+
+        private unsafe static (uint? userId, uint? groupId) GetUserAndGroupIds(string userName)
+        {
+            // First try with a buffer that should suffice for 99% of cases.
+            Interop.Sys.Passwd? passwd;
+            const int BufLen = 1024;
+            byte* stackBuf = stackalloc byte[BufLen];
+            if (TryGetPasswd(userName, stackBuf, BufLen, out passwd))
+            {
+                if (passwd == null)
+                {
+                    return (null, null);
+                }
+                return (passwd.Value.UserId, passwd.Value.GroupId);
+            }
+
+            // Fallback to heap allocations if necessary, growing the buffer until
+            // we succeed.  TryGetPasswd will throw if there's an unexpected error.
+            int lastBufLen = BufLen;
+            while (true)
+            {
+                lastBufLen *= 2;
+                byte[] heapBuf = new byte[lastBufLen];
+                fixed (byte* buf = &heapBuf[0])
+                {
+                    if (TryGetPasswd(userName, buf, heapBuf.Length, out passwd))
```

Comment:
>TryGetPasswd [](start = 24, length = 12)

Is it possible TryGetPasssword fail for other reasons than not enough buffer? if so, this can cause out of memory

Reply:
never mind. I see the implementation now so it return false only in case of not enough buffer

---
In reply to: [162512699](https://github.com/dotnet/corefx/pull/26431#discussion_r162512699) [](ancestors = 162512699)